### PR TITLE
improve codegen for golang code

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve? 
improve k8s codegen so that we don't need to do `hack/update-all.sh` in `$GOPATH` anymore!

### What is changed and how does it work?
change generator invocation script

### Check List 

Tests 

 - No code

Code changes

 - Has CI related scripts change

Side effects

None

Related changes

None

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -26,8 +26,12 @@ go mod vendor
...
 
+# `--output-base $SCRIPT_ROOT` will output generated code to current dir
 bash "${CODEGEN_PKG}"/generate-groups.sh "deepcopy,client,informer,lister" \
     github.com/pingcap/tidb-operator/pkg/client \
     github.com/pingcap/tidb-operator/pkg/apis \
     pingcap:v1alpha1 \
+    --output-base $SCRIPT_ROOT \
     --go-header-file ./hack/boilerplate/boilerplate.generatego.txt
+# then we merge generated code with our code base and clean up
+cp -r github.com/pingcap/tidb-operator/pkg $SCRIPT_ROOT && rm -rf github.com
```
Comment (DanielZhangQD) [inline hack/update-codegen.sh:37]: Does this work for both the cases where the code base is under $GOPATH or not?

Comment (DanielZhangQD) [reply hack/update-codegen.sh:37]: Could you please also help update the `CONTRIBUTING.md` to make it clear that if contributors update the CRD related types definition, they should run the `hack/update-all.sh`?

Comment (dragonly) [reply hack/update-codegen.sh:37]: https://github.com/pingcap/tidb-operator/pull/3519#discussion_r530263962

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**